util/network-devp2p/src/enr.rs

```diff
@@ -0,0 +1,434 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+// This module implements Ethereum Node Records as defined in EIP-778. A node record holds
```

Comment:
```suggestion
//! This module implements Ethereum Node Records as defined in [EIP-778](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-778.md). A node record holds
```

Module level doc comments start with `//!`.

Reply:
done